eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Repository/SearchIndex.php

```diff
@@ -1,266 +1,264 @@
     }
 
     /**
      * Get all words related to the content object (legacy db table: ezsearch_object_word_link).
-     *
-     * @param $contentId
-     *
-     * @return array
      */
-    public function getContentObjectWords($contentId)
+    public function getContentObjectWords(int $contentId): array
     {
-        $query = $this->dbHandler->createSelectQuery();
-
-        $this->setContentObjectWordsSelectQuery($query);
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->select('word_id')
+            ->from(self::SEARCH_OBJECT_WORD_LINK_TABLE)
+            ->where(
+                $query->expr()->eq(
+                    'contentobject_id',
+                    $query->createPositionalParameter($contentId, ParameterType::INTEGER)
+                )
+            );
 
-        $stmt = $query->prepare();
-        $stmt->execute(['contentId' => $contentId]);
-
-        $wordIDList = [];
-
-        while (false !== ($row = $stmt->fetch(PDO::FETCH_NUM))) {
-            $wordIDList[] = $row[0];
-        }
-
-        return $wordIDList;
+        return $query->execute()->fetchAll(FetchMode::COLUMN);
     }
 
     /**
      * Delete words not related to any content object.
      */
-    public function deleteWordsWithoutObjects()
+    public function deleteWordsWithoutObjects(): void
     {
-        $query = $this->dbHandler->createDeleteQuery();
-
-        $query->deleteFrom($this->dbHandler->quoteTable('ezsearch_word'))
-            ->where($query->expr->eq($this->dbHandler->quoteColumn('object_count'), ':c'));
-
-        $stmt = $query->prepare();
-        $stmt->execute(['c' => 0]);
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->delete(self::SEARCH_WORD_TABLE)
+            ->where(
+                $query->expr()->eq(
+                    'object_count',
+                    $query->createPositionalParameter(0, ParameterType::INTEGER)
+                )
+            );
+        $query->execute();
     }
 
     /**
      * Delete relation between a word and a content object.
-     *
-     * @param $contentId
      */
-    public function deleteObjectWordsLink($contentId)
+    public function deleteObjectWordsLink(int $contentId): void
     {
-        $query = $this->dbHandler->createDeleteQuery();
-        $query->deleteFrom($this->dbHandler->quoteTable('ezsearch_object_word_link'))
-            ->where($query->expr->eq($this->dbHandler->quoteColumn('contentobject_id'), ':contentId'));
-
-        $stmt = $query->prepare();
-        $stmt->execute(['contentId' => $contentId]);
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->delete(self::SEARCH_OBJECT_WORD_LINK_TABLE)
+            ->where(
+                $query->expr()->eq(
+                    'contentobject_id',
+                    $query->createPositionalParameter($contentId, ParameterType::INTEGER)
+                )
+            );
+        $query->execute();
```

Comment:
As far as I remember, `$query->execute()` will return `int` of rows affected. Return `void` is fine enough, just wonder wouldn't it be better to have `return $query->execute()` here. Same goes for other `delete*` method(s).

Reply:
Ok, no contract here, so seems harmless, fixed via 868d535fc